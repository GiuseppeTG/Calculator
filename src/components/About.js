import React from 'react';

export default function About() {
  return (
    <div className="about-section">
      <p className="quote">
        As far as the laws of mathematics refer to reality,
        they are not certain, and as far as they are certain,
        they do not refer to reality.
      </p>
      <p className="author">
        Richard Courant, German-American mathematician
      </p>
    </div>
  );
}
