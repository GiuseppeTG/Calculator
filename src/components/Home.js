import React from 'react';

export default function Home() {
  return (
    <div className="home-section">
      <h2 className="home-header">Welcome to our page!</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Aliquam amet consequuntur possimus, nulla id praesentium
        porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas,
        distinctio iste voluptates assumenda exercitationem obcaecati.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Aliquam amet consequuntur possimus, nulla id praesentium
        porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas,
        distinctio iste voluptates assumenda exercitationem obcaecati.
      </p>
    </div>
  );
}
