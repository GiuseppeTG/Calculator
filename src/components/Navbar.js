import React from 'react';
import { TbMathSymbols } from 'react-icons/tb';
import { Link } from 'react-router-dom';

export default function Navbar() {
  return (
    <nav>
      <div className="logo">
        <TbMathSymbols className="icon" />
        <p className="logo-text">
          Math Magician
        </p>
      </div>
      <ul className="links">
        <li><Link to="/" className="link">Home</Link></li>
        <li><Link to="/Calculator" className="link">Calculator</Link></li>
        <li><Link to="/about" className="link">About</Link></li>
      </ul>
    </nav>
  );
}
