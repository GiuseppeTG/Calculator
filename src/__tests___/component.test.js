import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../components/calculator';
import Home from '../components/Home';
import Navbar from '../components/Navbar';
import About from '../components/About';

describe('Test components', () => {
  it('Calculator test', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home test', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Navbar test', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('About test', () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
