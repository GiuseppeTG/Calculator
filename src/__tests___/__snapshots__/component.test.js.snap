// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components Calculator test 1`] = `
<div
  className="calculator-section"
>
  <h2>
    Let's do some Math
  </h2>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      0
    </div>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calc-btn 
                                   
                                  operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calc-btn 
                                   
                                  operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calc-btn 
                                   
                                  operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calc-btn 
                                   
                                  operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="calc-btn 
                                  span-two 
                                  "
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="calc-btn 
                                   
                                  "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calc-btn 
                                   
                                  operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Test components Home test 1`] = `
<div
  className="home-section"
>
  <h2
    className="home-header"
  >
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem obcaecati.
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem obcaecati.
  </p>
</div>
`;
